x-pack/plugins/observability_solution/inventory/common/entities.ts

```diff
@@ -102,23 +99,22 @@ export const entityTypesRt = new t.Type<string[], string, unknown>(
   (arr) => arr.join()
 );
 
-export interface Entity {
-  [ENTITY_LAST_SEEN]: string;
-  [ENTITY_ID]: string;
-  [ENTITY_TYPE]: string;
-  [ENTITY_DISPLAY_NAME]: string;
-  [ENTITY_DEFINITION_ID]: string;
-  [ENTITY_IDENTITY_FIELDS]: string | string[];
-  alertsCount?: number;
-  [key: string]: any;
-}
-
 export type EntityGroup = {
   count: number;
 } & {
   [key: string]: any;
 };
 
-export type InventoryEntityLatest = z.infer<typeof entityLatestSchema> & {
+export type EntityLatest = z.infer<typeof entityLatestSchema>;
+
+export type InventoryEntityLatest = {
+  entityId: string;
+  entityType: string;
+  entityIdentityFields: string | string[];
+  entityDisplayName: string;
+  entityDefinitionId: string;
+  entityLastSeenTimestamp: string;
+  entityDefinitionVersion: string;
+  entitySchemaVersion: string;
   alertsCount?: number;
-};
+} & z.infer<typeof entityMetadataSchema>;
```

Comment:
The idea here is to make the API return an interface that doesn't depend in the index mapping.